### DISCUSSION TITLE
How to set themeColor with .NET AdaptiveCards SDK

### DISCUSSION THREAD

**Discussion by soulflyman** [microsoft/AdaptiveCards#6382] [Q&A] [answered]
When sending the following json to a MS Teams Webhook I can change the color bar above the message with the `themeColor` Option:
```json
{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "MessageCard",    
    "themeColor": "ff0000",
    "version": "1.2",
    "title": "Alarm!",
    "text": "Test Message" 
}
```

![2021-09-21 17_14_24-Microsoft Teams](https://user-images.githubusercontent.com/379492/134198261-2780dd0a-0a8f-486a-b2bf-67870ae184bb.png)

How is this possible with the .NET AdaptiveCards SDK that ist available via nuget?

**Comment by soulflyman** [comment, marked as answer]
After creating an Issue from my question (#6841) it was forwarded to the MS-Teams tracker [msteams-docs/#4854](https://github.com/MicrosoftDocs/msteams-docs/issues/4854) an they hinted me to use the O365ConnectorCard Class [[1]](https://docs.microsoft.com/en-us/dotnet/api/microsoft.bot.schema.teams.o365connectorcard?view=botbuilder-dotnet-stable) [[2]](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/task-modules-and-cards/cards/cards-reference.md#office-365-connector-card).

**Reply by qris** [reply]
Unfortunately Office 365 Connectors are now [deprecated and being shut down](https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/), so it would be good to have a real solution to this.

**Reply by MichelZ** [reply]
We are using "normal" adaptive cards, and have achieved this more or less using the following technique:
- Use "backgroundImage" on the card
- Set the image to 1x1 or 1x2 pixel image in the color you want
- Set fillMode to RepeatHorizontally

![image](https://github.com/user-attachments/assets/779e8ad2-56f2-44dc-b48e-461ca8ec359e)


```
{
  "type": "AdaptiveCard",
  "body": [
    { }
  ],
  "actions": [
    {
     
    }
  ],
  "msteams": {
    "width": "Full"
  },
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.5",
  "backgroundImage": {
    "url": "https://ourpixelgenerator/api/pixel/${accentColor}/2",
    "fillMode": "RepeatHorizontally"
  }
}
```

As you can see we're using a simple API to generate the pixel in the color we need and in the height we need it

Hope that helps and is a way that is and stays supported going forward with all the O365 connector stuff going away
